src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt

```diff
@@ -155,6 +155,26 @@ class ProjectsController(private val session: Session,
         return SuccessResponse(currentProject).asResponseEntity()
     }
 
+    //  fun promoteVersion(
+    //     @PathVariable("projectId") projectId: Int,
+    //     @PathVariable("versionId") versionId: String,
+    //     @RequestParam("name") name: String): ResponseEntity<String>
+    // {
+
+    @PostMapping("/project/{projectId}/rename")
+    @ResponseBody
+    fun renameProject(
+        @PathVariable("projectId") projectId: Int,
+        @RequestParam("name") name: String): ResponseEntity<String>
+    {
+        projectRepository.renameProject(projectId, userId(), name)
+        val currentProject = projectRepository.getProjectFromVersionId(session.getVersionId(), userId())
+        if (currentProject.id == projectId){
+            session.setVersionId(null)
```

Comment:
Alternatively we could have it simply re-request the currentVersion/Project rather than nulling it so that it updates to the new name it has been given.

Reply:
That probably would be slightly nicer!